pcdsdevices/signal.py

```diff
@@ -116,30 +122,61 @@ class FakePytmcSignalRO(FakePytmcSignal, FakeEpicsSignalRO):
 fake_device_cache[PytmcSignal] = FakePytmcSignal
 
 
+def is_meek_ophyd_object(obj: ophyd.ophydobj.OphydObject):
+    """
+    Can you expect the ophyd object to ever give subscription callbacks?
+
+    If not, pcdsdevices considers that object to be "meek" or quiet.  This can
+    be useful to determine whether or not you should wait on a subscription
+    callback to happen before continuing.
+    """
+    if not isinstance(obj, ophyd.ophydobj.OphydObject):
+        raise ValueError(f"{obj} is not an ophyd object")
+
+    return any(isinstance(obj, cls) for cls in fake_device_cache.values())
+
+
+@dataclasses.dataclass
+class _AggregateSignalState:
+    """
+    This class holds per-Signal state information when used as part of an
+    AggregateSignal.
+
+    It includes a cache of the last value, connectivity status, and callback
+    identifiers from ophyd.
+    """
+    #: The signal itself
+    signal: Signal
+    #: Is the signal connected according to its metadata callback?
+    connected: bool = False
+    #: The last value retrieved from a value callback, or a direct get request.
+    value: Optional[OphydDataType] = None
+    #: The value subscription callback ID (None if not yet subscribed)
+    value_cbid: Optional[int] = None
+    #: The meta subscription callback ID (None if not yet subscribed)
+    meta_cbid: Optional[int] = None
+
+
 class AggregateSignal(Signal):
```

Comment:
It seems like the edits to AggregateSignal are primarily aimed at cleaning up the rough edges of the class and adding proper support for metadata. Is my understanding correct, or did I miss/misunderstand something?

Reply:
That is correct. The PR started off as a light change to the class but gradually became more thorough, as these things tend to go.